src/renderer/api/kube-watch-api.ts

```diff
@@ -1,201 +1,326 @@
-// Kubernetes watch-api consumer
+// Kubernetes watch-api client
+// API: https://developer.mozilla.org/en-US/docs/Web/API/Streams_API/Using_readable_streams
 
-import { computed, observable, reaction } from "mobx";
-import { stringify } from "querystring";
-import { autobind, EventEmitter } from "../utils";
-import { KubeJsonApiData } from "./kube-json-api";
+import type { Cluster } from "../../main/cluster";
+import type { IKubeWatchEvent, IKubeWatchEventStreamEnd, IWatchRoutePayload } from "../../main/routes/watch-route";
+import type { KubeObject } from "./kube-object";
 import type { KubeObjectStore } from "../kube-object.store";
-import { ensureObjectSelfLink, KubeApi } from "./kube-api";
+import type { NamespaceStore } from "../components/+namespaces/namespace.store";
+
+import debounce from "lodash/debounce";
+import { comparer, computed, observable, reaction } from "mobx";
+import { autobind, EventEmitter } from "../utils";
+import { ensureObjectSelfLink, KubeApi, parseKubeApi } from "./kube-api";
+import { KubeJsonApiData, KubeJsonApiError } from "./kube-json-api";
+import { apiPrefix, isProduction } from "../../common/vars";
 import { apiManager } from "./api-manager";
-import { apiPrefix, isDevelopment } from "../../common/vars";
-import { getHostedCluster } from "../../common/cluster-store";
 
-export interface IKubeWatchEvent<T = any> {
-  type: "ADDED" | "MODIFIED" | "DELETED" | "ERROR";
-  object?: T;
+export { IKubeWatchEvent, IKubeWatchEventStreamEnd };
+
+export interface IKubeWatchMessage<T extends KubeObject = any> {
```

Comment:
```suggestion
export interface IKubeWatchMessage<T extends KubeObject = KubeObject> {
```

`any` is not `KubeObject`